deltacat/compute/converter/steps/convert.py

```diff
@@ -34,16 +34,27 @@ def convert(convert_input: ConvertInput):
 
     logger.info(f"Starting convert task index: {convert_task_index}")
     data_files, equality_delete_files, position_delete_files = files_for_each_bucket[1]
+    # Get string representation of partition value out of Record[partition_value]
+    partition_value_str = (
+        files_for_each_bucket[0].__repr__().split("[", 1)[1].split("]")[0]
```

Comment:
The code to extract partition values looks pretty unfortunately opaque (but I understand this happens semi-regularly, esp. when trying to access details like this that Iceberg prefers to hide from external users). Maybe we can at least put these one-liners inside of appropriately named helper methods?